### DISCUSSION TITLE
Flutter + Supabase: How to Handle and Monitor Complex Event Chains ?

### DISCUSSION THREAD

**Discussion by Felixburton7** [supabase/supabase#32417] [Questions]
Hi everyone,  

I’m building an app with a workflow that triggers a chain of interconnected events based on user actions and data from an external API. Here’s the flow:  

1. A user selects a team.  
2. If the team wins a match (result fetched from an external API), they are removed from the league (marked as `FALSE` in the league table).  
3. Based on the outcome, money is allocated to users, and other updates occur across multiple tables (e.g., updating user balances, recording match results, modifying league statuses, and applying changes to other user tables/columns).  
4. Supabase triggers and functions handle much of this backend logic, including updating league tables, recalculating rankings, and sending notifications.  

Here’s where I’m running into challenges:  
- **Debugging**: When the flow breaks or behaves unexpectedly, it’s tough to figure out where the issue lies—whether it’s a problem with a trigger, a function, or an external API input.  
- **Monitoring**: I don’t have a clear way to track these cascading events in real time. Understanding how each update impacts the others, especially when external data kicks off the process, can be tricky.  
- **Scaling**: The complexity of this event chain is growing as I add new features. I want to ensure the system remains clean, performant, and maintainable over time.  

If you’ve faced similar challenges or have advice on tools, workflows, or best practices for managing event chains in Supabase, I’d love to hear your insights!  

Thanks in advance for your help! 🙌

**Comment by wjones2810** [comment]
Following this^
